toonz/sources/toonz/xshtoolbar.cpp

```diff
@@ -0,0 +1,265 @@
+  ret = ret && connect(m_reframe2sButton, SIGNAL(released()), this,
+                       SLOT(onReframe2sButtonPressed()));
+  ret = ret && connect(m_reframe3sButton, SIGNAL(released()), this,
+                       SLOT(onReframe3sButtonPressed()));
+  ret = ret && connect(m_repeatButton, SIGNAL(released()), this,
+                       SLOT(onRepeatButtonPressed()));
+  ret = ret && connect(m_collapseSubButton, SIGNAL(released()), this,
+                       SLOT(onCollapseSubButtonPressed()));
+  ret = ret && connect(m_enterSubButton, SIGNAL(released()), this,
+                       SLOT(onEnterSubButtonPressed()));
+  ret = ret && connect(m_leaveSubButton, SIGNAL(released()), this,
+                       SLOT(onLeaveSubButtonPressed()));
+
+  assert(ret);
+
+  m_leaveSubButton->hide();
+}
+
+//-----------------------------------------------------------------------------
+
+void Toolbar::onNewVectorLevelButtonPressed() {
+  int defaultLevelType = Preferences::instance()->getDefLevelType();
+  Preferences::instance()->setDefLevelType(PLI_XSHLEVEL);
+  CommandManager::instance()->execute("MI_NewLevel");
+  Preferences::instance()->setDefLevelType(defaultLevelType);
+}
+
+//-----------------------------------------------------------------------------
+
+void Toolbar::onNewToonzRasterLevelButtonPressed() {
+  int defaultLevelType = Preferences::instance()->getDefLevelType();
+  // Preferences::instance()->setOldDefLevelType(defaultLevelType);
+  Preferences::instance()->setDefLevelType(TZP_XSHLEVEL);
+  CommandManager::instance()->execute("MI_NewLevel");
+  Preferences::instance()->setDefLevelType(defaultLevelType);
+}
+
+//-----------------------------------------------------------------------------
+
+void Toolbar::onNewRasterLevelButtonPressed() {
+  int defaultLevelType = Preferences::instance()->getDefLevelType();
+  Preferences::instance()->setDefLevelType(OVL_XSHLEVEL);
+  CommandManager::instance()->execute("MI_NewLevel");
+  Preferences::instance()->setDefLevelType(defaultLevelType);
+}
+
+//-----------------------------------------------------------------------------
+
+void Toolbar::onReframe1sButtonPressed() {
+  CommandManager::instance()->execute("MI_Reframe1");
+}
+
+//-----------------------------------------------------------------------------
+
+void Toolbar::onReframe2sButtonPressed() {
+  CommandManager::instance()->execute("MI_Reframe2");
+}
+
+//-----------------------------------------------------------------------------
+
+void Toolbar::onReframe3sButtonPressed() {
+  CommandManager::instance()->execute("MI_Reframe3");
+}
+
+//-----------------------------------------------------------------------------
+
+void Toolbar::onRepeatButtonPressed() {
+  CommandManager::instance()->execute("MI_Dup");
+}
+
+//-----------------------------------------------------------------------------
+
+void Toolbar::onCollapseSubButtonPressed() {
+  CommandManager::instance()->execute("MI_Collapse");
```

Comment:
In most of the slots you are just invoking a correspondent action from `CommandManager` .
How about to make the toolbar buttons by `QToolBar->addAction(QAction*)` instead of `addWidget` for such case?

You can set icon and tooltip for these actions in `mainwindow.cpp` (Please refer my modification [here](https://github.com/opentoonz/opentoonz/pull/1060/files#diff-c4e8720ba98e1879d0179b7851697c42).)
Buttons can be styled like as follows: 
```
#XSheetToolbar {
	margin: 0;
	padding: 0;
	border: 0;
		&::separator:horizontal {
		margin-right: 7;
		margin-left: 5;
	}
	& QToolButton {
		.base_outset;
		border-width: 1px;
		border-radius: 4px;
		margin: 2;
		padding: 3;
		&:pressed {
			border-image: none;
			.base_inset(light, 10%);
		}
		/*lighten lilttle bit when hover*/
		&:hover {
			border-image: none;
			.base_outset(light, 10%);
			&:pressed {
				.base_inset(light, 10%);
			}
		}
		/*lighten lilttle bit when pressed*/
		&:disabled{
			.base_outset(light, 5%);
			color: rgb(80,80,80);
		}
	}
}
```
IMO It will improve UI consistency in two points: First, the same icons will be added to the main menu bar as well. Secondly, the actions' state (enable/disable) will be reflected to the tool bar buttons.